site/en/docs/extensions/reference/declarativeContent/index.md

```diff
@@ -2,26 +2,31 @@
 api: declarativeContent
 ---
 
+{% Aside %}
+
+To transition from pageaction to action, see [Emulating pageActions with declarativeContent][emulating-page-actions]
+
+{% endAside %}
+
 ## Usage
 
-The Declarative Content API allows you to show your extension's [page action][1] depending on the
-URL of a web page and the CSS selectors its content matches, without needing to take a [host
-permission][2] or inject a [content script][3]. Use the [activeTab][4] permission in order to be
-able to interact with a page after the user clicks on your page action.
+The Declarative Content API allows you to enable your extension's [action][api-action] depending on
+the URL of a web page and the CSS selectors its content matches, without needing to add [host
+permission][docs-host-perm] or inject a [content script][docs-content-scripts]. 
 
-If you need more precise control over when your page action appears or you need to change its
-appearance to match the current tab before the user clicks on it, you'll have to keep using the
-[pageAction][5] API.
+Use the [activeTab][docs-activetab] permission to interact with a page after the user clicks on the
+extension's action (toolbar icon).
 
 ## Rules
 
-As a [declarative API][6], this API lets you register rules on the [`onPageChanged`][7] [event][8]
-object which take an action ([`ShowPageAction`][9] and [`SetIcon`][10]) when a set of conditions,
-represented as a [`PageStateMatcher`][11], are met.
+As a [declarative API][api-declarative], this API lets you register rules on the
+[`onPageChanged`][event-onpagechanged] event object which take an action
```

Comment:
>lets you register rules on the on the `onPageChanged` event object

This phrase feels like its cramming too much in formation in too small a space and I can't disentangle it. I would expect something called `onPageChanged` to be defined on a target object, take a custom method (a callback, though it's not usually called that in JS) and for the method to get passed an event object. This description reads like it's conflated the target, the callback, and the event into one thing. It's confusing.

Reply:
Agreed! I restructured this section to detangle key concepts. Let me know what you think.